packages/ast/src/encoding/proto/types.ts

```diff
@@ -232,6 +240,10 @@ export const getFieldOptionalityForDefaults = (
     const useOptionalNullable = context.pluginValue('prototypes.useOptionalNullable');
     const isScalarDefaultToNullable = !field?.options?.['(gogoproto.nullable)'] && context.pluginValue('prototypes.isScalarDefaultToNullable');
 
+    if(field?.options?.['(gogoproto.moretags)'] && getOptionDeprecated(field?.options?.['(gogoproto.moretags)']) === 'false'){
```

Comment:
here too

correct logic here should be:

    if(field?.options?.['(gogoproto.moretags)'] && getOptionDeprecated(field?.options?.['(gogoproto.moretags)']) === 'true'){
        return true
    }